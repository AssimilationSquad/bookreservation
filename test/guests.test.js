import { shallow, mount, render } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Guests from '../client/guests.jsx';
import Calendar from '../client/calendar.jsx';


describe('<Guests />', () => {
  it('renders 6 <div> components', () => {
    const wrapper = shallow(<Guests />);
    expect(wrapper.find('div')).toHaveLength(6);
  });

  it('looks the same as before', () => {
    const tree = renderer.create(<Guests />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('<Calendar />', () => {
  it('renders a calendar', () => {
    const wrapper = shallow(<Calendar year="2018" month="10" handleMonthChange={() => {}} />);
    expect(wrapper.find('td').length).toBeGreaterThan(30);
  });
});