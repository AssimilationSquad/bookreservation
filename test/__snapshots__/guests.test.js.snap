// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Guests /> looks the same as before 1`] = `
<div
  id="guests-box"
>
  <div
    className="age-group"
    id="adults"
  >
    <span
      className="age-left"
    >
      Adults
    </span>
    <span
      className="age-right"
    >
      <button
        className="circle minus"
      >
        -
      </button>
      <span>
        1
      </span>
      <button
        className="circle plus"
      >
        +
      </button>
    </span>
  </div>
  <div
    className="age-group"
    id="children"
  >
    <span
      className="age-left"
    >
      Children
    </span>
    <span
      className="age-right"
    >
      <button
        className="circle minus"
      >
        -
      </button>
      <span>
        0
      </span>
      <button
        className="circle plus"
      >
        +
      </button>
    </span>
  </div>
  <div
    className="age-group"
    id="infants"
  >
    <span
      className="age-left"
    >
      Infants
    </span>
    <span
      className="age-right"
    >
      <button
        className="circle minus"
      >
        -
      </button>
      <span>
        0
      </span>
      <button
        className="circle plus"
      >
        +
      </button>
    </span>
  </div>
  <div
    id="max-guests"
  >
    3 guests maximum. Infants donâ€™t count toward the number of guests.
  </div>
  <div
    id="close"
  >
    <span>
      Close
    </span>
  </div>
</div>
`;
